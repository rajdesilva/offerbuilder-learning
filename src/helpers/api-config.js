import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});