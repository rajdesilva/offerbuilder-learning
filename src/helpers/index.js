export * from './api-config';
export * from './history';
