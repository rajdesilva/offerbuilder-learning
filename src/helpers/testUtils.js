import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
