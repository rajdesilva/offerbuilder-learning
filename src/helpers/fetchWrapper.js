import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
