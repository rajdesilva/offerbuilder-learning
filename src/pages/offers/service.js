import { baseURL } from "../../helpers";
import {
  getParsedOffers,
  getSearchAndFilterImagesQueryUrl,
  getSearchAndFilterQueryURL,
  updateReduxWithOfferData,
} from "../../helpers/utility";
import { store } from "../../redux/store";
import {
  marketingActions,
  newOfferActions,
  offerListSearchAndFilterActions,
} from "./actions";
import { offerImageSearchAndFilterActions } from "./actions/offerImageSearchAndFilterActions";

export const validateOfferId = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/exist`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCloudinarySignature = () => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}cloudinary/signature`)
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error.toString();
    });
};

export const getCityFromLatLong = (latLong) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latLong.lat},${latLong.lng}&sensor=true&key=AIzaSyDydZJ7RZF1SAXwg4pZqnFFskYL8CW5RqA`;
  return fetch(url)
    .then((response) => response.json())
    .then((response) => {
      return response;
    })
    .catch((error) => {
      return error;
    });
};

export const getLanguages = () => {
  store.dispatch({
    type: marketingActions.FETCH_LANGUAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}languages`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_SUCCESS,
          payload: window.getValue(response, "data.languages"),
        });
      } else {
        store.dispatch({
          type: marketingActions.FETCH_LANGUAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: marketingActions.FETCH_LANGUAGE_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const createNewOffer = (newOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers`;
  store.dispatch({
    type: newOfferActions.CREATE_NEW_OFFER_LOADING,
    payload: true,
  });
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const saveEditedOffer = (updatedOfferInfo) => {
  const url = `${baseURL.API_URL_BASE}offers/${updatedOfferInfo.id}`;
  return window
    .fetchWrapper(url, {
      method: "PATCH",
      body: JSON.stringify(updatedOfferInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const getOfferDetails = (offerId) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_DETAILS_LOADING,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`)
    .then((response) => {
      if (response.success) {
        let offerData = window.getValue(response, "data.offer");
        offerData = getParsedOffers(offerData);
        updateReduxWithOfferData(offerData);
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_SUCCESS,
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
        });
      }
    })
    .catch((error) => {
      console.log(error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_DETAILS_FAILURE,
      });
    });
};

export const deleteOffer = (offerId) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}`, {
      method: "DELETE",
    })
    .then((response) => response);
};

export const updateOffer = (offerId, status) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}offers/${offerId}/${status}`, {
      method: "PUT",
    })
    .then((response) => response);
};

export const searchAndFilterOffer = (offerType) => {
  store.dispatch({
    type: offerListSearchAndFilterActions.OFFER_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}offers/search/` +
        getSearchAndFilterQueryURL(offerType)
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerListSearchAndFilterActions.OFFER_LIST_FAILURE,
        payload: [],
      });
    });
};

export const getImagesFromProperties = (propertyCode) => {
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}search?propertyCode=${propertyCode}`)
    .then((response) => response);
};

export const searchAndFilterImages = () => {
  store.dispatch({
    type: offerImageSearchAndFilterActions.OFFER_IMAGE_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}images/search` +
        getSearchAndFilterImagesQueryUrl()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      console.error("error =", error.toString());
      store.dispatch({
        type: offerImageSearchAndFilterActions.OFFER_IMAGE_FAILURE,
        payload: [],
      });
    });
};

export const uploadImage = (file) => {
  let myHeaders = new Headers();

  myHeaders.append("Accept", "application/json");
  let formdata = new FormData();
  formdata.append("file", file);
  return window
    .fetchWrapper(`${baseURL.API_URL_BASE}images/upload`, {
      method: "POST",
      headers: myHeaders,
      body: formdata,
    })
    .then((response) => response);
};
