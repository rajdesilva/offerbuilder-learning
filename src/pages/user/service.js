import { baseURL } from "../../helpers";
import { userActions } from "./actions/userActions";
import { store } from "../../redux/store";

export const fetchUserInfo = () => {
  const url = `${baseURL.API_URL_BASE}user/`;
  store.dispatch({
    type: userActions.FETCH_USER_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(url)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_FAILURE,
          payload: false,
        });
      }
    })
    .catch(() => {
      store.dispatch({
        type: userActions.FETCH_USER_FAILURE,
        payload: false,
      });
    });
};

export const getUsersList = () => {
  store.dispatch({
    type: userActions.FETCH_USER_MANAGEMENT_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}users/`
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
        payload: [],
      });
    });
};

export const addNewUser = (newUserInfo) => {
  const url = `${baseURL.API_URL_BASE}users/user`;
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newUserInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const deleteUser = (email, userRole) => {
  const url = `${baseURL.API_URL_BASE}users/user/${email}?roleNames=${userRole}`;
  return window
    .fetchWrapper(url, {
      method: "DELETE",
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};
