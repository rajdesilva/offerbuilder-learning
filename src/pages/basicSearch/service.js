import { browseSupplyActions, supplySearchActions } from "./actions";
import { store } from "./../../redux/store";
import { globalActions } from "../../redux/globalActions";
import { baseURL } from "../../helpers";
import {
  createSearchQueryURL,
  getSearchParamsToFetchMarketPrice,
} from "../../helpers/utility";
import { offerListSearchAndFilterActions } from "../offers/actions";

export const searchSupply = (routePostSearch, isEditFlow) => {
  store.dispatch({
    type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
    payload: true,
  });
  store.dispatch({
    type: supplySearchActions.SET_INCLUDE_MARKET_PRICE_FLAG,
    payload: false,
  });

  window
    .fetchWrapper(`${baseURL.API_URL_BASE}search/` + createSearchQueryURL(routePostSearch))
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_SUCCESS,
          payload: window.getValue(response, "data.properties"),
          totalMatch: window.getValue(response, "data.totalMatchedProperties"),
          pageOffset: window.getValue(response, "data.pageOffset"),
          pageSize: window.getValue(response, "data.pageSize"),
        });
      } else {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
          payload: window.getValue(response, "error.apierror.message"),
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
        payload: error.toString(),
      });
      store.dispatch({
        type: globalActions.SET_ERROR,
        payload: error.toString(),
      });
    });
};

export const advancedSearch = (
  params,
  searchType,
  page,
  pageSize,
  sortBy,
  order,
  groupType
  ) => {
    return window.fetchWrapper(`${baseURL.API_URL_BASE}global/properties?address=${params.address}&city=${params.city}&countryCode=${params.countryCode}&distance=${params.distance}&hotelName=${params.hotelName}&latitude=${params.latitude}&longitude=${params.longitude}&zip=${params.zip}&pageOffset=${page}&pageSize=${pageSize}&propertyGroupType=${groupType}&sortCriteria=${sortBy}&sortOrder=${order}&globalSearchType=${searchType}`);

    //return window.fetchWrapper(`https://offerbuilder-api-dev.kognitiv.com/global/properties?distance=10&globalSearchType=BASIC&latitude=47.2816&longitude=15.48809&propertyGroupType=ALL`);
}

export const globalSearchResult =(params)=>{
  
  window
    .fetchWrapper(`/global/properties?address=${params.address}`)
    .then((response) => {
        console.log(response)
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
}

export const getChannels = () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_CHANNELS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}portals`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_SUCCESS,
          payload: window.getValue(response, "data.result"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getBrands = async () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_BRANDS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}sources`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_SUCCESS,
          payload: window.getValue(response, "data.brands"),
        });
        store.dispatch({
          type: offerListSearchAndFilterActions.UPDATE_OFFER_SELECTORS,
          payload: window.getValue(response, "data.brands"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_FAILURE,
          payload: [],
        });
      }
      return window.getValue(response, "data.brands");
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_BRANDS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getCurrencies = async () => {
  store.dispatch({
    type: supplySearchActions.FETCH_CURRENCIES_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}currencies`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_SUCCESS,
          payload: window.getValue(response, "data.currencies"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getMarketPricesForListOfProperties = () => {
  store.dispatch({
    type: supplySearchActions.FETCH_MARKET_PRICE_LOADING,
    payload: true,
  });

  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}search/marketprice/` +
        getSearchParamsToFetchMarketPrice()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
        payload: [],
      });
      return [];
    });
};
