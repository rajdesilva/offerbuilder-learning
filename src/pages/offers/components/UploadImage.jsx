import React, { useState, Fragment } from "react";
import { InboxOutlined } from "@ant-design/icons";
import { Button, message, Upload, Modal } from "antd";
import { FormattedMessage } from "react-intl";
import { uploadImage } from "../service";
import styles from "./css/UploadImage.module.less";

export default function UploadImage(props) {
  const [loading, setLoading] = useState(false);
  const [fileList, setFileList] = useState([]);
  const [uploadedFiles, setUplodedFiles] = useState([]);

  const handleOnChange = () => {
    setLoading(true);
    try {
      fileList.forEach(async (file, index) => {
        uploadImage(file)
          .then((response) => {
            if (window.getValue(response, "success")) {
              message.success(file.name + " uploaded successfully!");
              uploadedFiles.push(window.getValue(response, "data"));

              setUplodedFiles(uploadedFiles);
              if (uploadedFiles.length === fileList.length) {
                props.toggleUploadModal(false, uploadedFiles);
              }
            } else {
              if (window.getValue(response, "error.message")) {
                message.error(window.getValue(response, "error.message"));
              } else {
                message.error(
                  file.name + " failed to upload. Please try again"
                );
              }
            }
            setLoading(false);
          })
          .catch((error) => {
            setLoading(false);
            console.log("error for file =", file, " error =", error.toString());
            message.error(error.toString());
          });
      });
    } catch (e) {
      setLoading(false);
      message.error(e.toString());
    }
  };

  return (
    <Modal
      title={<FormattedMessage id="nemo.uploadForLocal" />}
      visible={true}
      closable={true}
      width={1080}
      centered
      destroyOnClose={true}
      className={
        fileList && fileList.length === 0
          ? styles.propertyWithNoImageModal
          : styles.propertyImageModal
      }
      okText={<FormattedMessage id="nemo.submit" />}
      onCancel={() => props.toggleUploadModal(false)}
      data-testid="image-upload-for-offer-modal"
      footer={[
        <Button
          key="cancel"
          onClick={() => {
            props.toggleUploadModal(false);
          }}
          data-testid="upload-image-cancel-btn"
        >
          <FormattedMessage id="nemo.cancel" />
        </Button>,
        <Button
          key="submit"
          type="primary"
          loading={loading}
          disabled={fileList && fileList.length === 0}
          onClick={handleOnChange}
          data-testid="image-upload-btn"
        >
          <FormattedMessage id="nemo.upload" />
        </Button>,
      ]}
    >
      <Fragment>
        <Upload.Dragger
          name="file"
          accept="image/png, image/jpeg"
          fileList={fileList}
          data-testid="image-file-upload-dragger"
          onRemove={(file) => {
            const index = fileList.indexOf(file);
            const newFileList = fileList.slice();
            newFileList.splice(index, 1);
            setFileList(newFileList);
          }}
          onChange={({ localFile, localFileList }) => {
            if (localFileList) {
              setFileList(localFileList.map((file) => file.originFileObj));
            }
          }}
          beforeUpload={(file, fileList) => {
            setFileList(fileList);
            return false;
          }}
          multiple={true}
        >
          <p className="ant-upload-drag-icon">
            <InboxOutlined />
          </p>
          <p className="ant-upload-text">
            <FormattedMessage id="nemo.clickOrDragToUpload" />
          </p>
          <p className="ant-upload-hint">
            <FormattedMessage id="nemo.supportForSingleOrBulkUpload" />
          </p>
        </Upload.Dragger>
      </Fragment>
    </Modal>
  );
}
