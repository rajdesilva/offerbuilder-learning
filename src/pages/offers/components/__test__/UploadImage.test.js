import React from "react";
import { createStore } from "redux";
import { render, fireEvent, cleanup } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import UploadImage from "../UploadImage";
import * as service from "./../../service";
import { tempStore } from "./tempStore";
import { act } from "react-test-renderer";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});
jest.mock("./../../service");
afterEach(cleanup);

describe("UploadImage component test", () => {
  test("create UploadImage components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      toggleUploadModal: jest.fn(),
    };
    const { container } = render(
      <MemoryRouter>
        <UploadImage {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(container).toMatchSnapshot();
  });

  test("UploadImage click on close btn to hide upload view", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      toggleUploadModal: jest.fn(),
    };
    const { queryByTestId, getByTestId } = render(
      <MemoryRouter>
        <UploadImage {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );

    const cancelBtn = getByTestId("upload-image-cancel-btn");
    expect(cancelBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.click(cancelBtn);
      expect(props.toggleUploadModal).toHaveBeenCalledTimes(1);
      expect(
        queryByTestId("image-upload-for-offer-modal")
      ).not.toBeInTheDocument();
    });
  });

  test("UploadImage click on upload btn and service is not called as filelist is 0", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      toggleUploadModal: jest.fn(),
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <UploadImage {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );

    const uploadBtn = getByTestId("image-upload-btn");
    expect(uploadBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.click(uploadBtn);
      expect(props.toggleUploadModal).toHaveBeenCalledTimes(0);
    });
  });

  test("UploadImage click on upload btn and service is called with file selected to upload and successful", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      toggleUploadModal: jest.fn(),
    };
    const { queryByTestId, getByTestId } = render(
      <MemoryRouter>
        <UploadImage {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    service.uploadImage.mockImplementation((data) => {
      return Promise.resolve({
        success: true,
        data: {
          id: "chucknorris.png",
          uploadedBy: "swapnil.deshmukh@kognitiv.com",
          checksum: "7634f719f11b8d5673b85b3707029863",
          url: "https://res.cloudinary.com/seekda-dev/image/upload/v1616688107/offer_builder_dev/chucknorris.png",
          createdAt: "2021-03-25T17:01:47.516244",
          updatedAt: "2021-03-25T17:01:47.516244",
        },
        error: null,
      });
    });
    const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
    const inputEl = getByTestId("image-file-upload-dragger");
    Object.defineProperty(inputEl, "files", {
      value: [file],
    });
    const uploadBtn = getByTestId("image-upload-btn");
    expect(uploadBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.change(inputEl);
      await fireEvent.click(uploadBtn);
      expect(props.toggleUploadModal).toHaveBeenCalledTimes(1);
      expect(
        queryByTestId("image-upload-for-offer-modal")
      ).not.toBeInTheDocument();
    });
  });

  test("UploadImage click on upload btn and service is called with file selected to upload and failed", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const props = {
      toggleUploadModal: jest.fn(),
    };
    const { getByTestId } = render(
      <MemoryRouter>
        <UploadImage {...props} />
      </MemoryRouter>,
      {
        store,
      }
    );
    service.uploadImage.mockImplementation((data) => {
      return Promise.reject({
        success: false,
        data: null,
        error: null,
      });
    });
    const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
    const inputEl = getByTestId("image-file-upload-dragger");
    Object.defineProperty(inputEl, "files", {
      value: [file],
    });
    const uploadBtn = getByTestId("image-upload-btn");
    expect(uploadBtn).toBeTruthy();
    await act(async () => {
      await fireEvent.change(inputEl);
      await fireEvent.click(uploadBtn);
      expect(props.toggleUploadModal).toHaveBeenCalledTimes(0);
    });
  });
});
