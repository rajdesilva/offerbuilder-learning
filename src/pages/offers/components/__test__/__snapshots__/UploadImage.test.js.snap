// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadImage component test create UploadImage components snapshot 1`] = `<div />`;
